website/static/js/contribManager.js

```diff
@@ -6,106 +6,127 @@ var bootbox = require('bootbox');
+
+    self.visible = ko.observable(contributor.visible);
+
+    self.permission.subscribeChanged(function(newValue, oldValue) {
+        if (oldValue === 'admin') {
+            parent.adminCount(parent.adminCount() - 1);
+        }
+        if (newValue === 'admin') {
+            parent.adminCount(parent.adminCount() + 1);
+        }
+        self.permissionText(parent.permissionDict[newValue]);
+    });
+
+    self.visible.subscribeChanged(function(newValue, oldValue) {
+        if (oldValue === true) {
+            parent.visibleCount(parent.visibleCount() - 1);
+        }
+        if (newValue === true) {
+            parent.visibleCount(parent.visibleCount() + 1);
+        }
+    });
+
+    self.permissionChange = ko.computed(function() {
+        return self.permission() !== self.originals.permission;
+    });
+
+    self.reset = function() {
+        if (self.deleteStaged()) {
+            if (self.visible()) {
+                parent.visibleCount(parent.visibleCount() + 1);
             }
+            if (self.permission() === 'admin') {
+                parent.adminCount(parent.adminCount() + 1);
+            }
+            self.deleteStaged(false);
         }
-        return self.permissionList[0];
+        self.permission(self.originals.permission);
+        self.visible(self.originals.visible);
+        self.index(self.originals.index);
     };
 
+    self.index = ko.observable(index);
+
     self.currentUserCanEdit = currentUserCanEdit;
     self.isAdmin = isAdmin;
-    self.visible = ko.observable(contributor.visible);
-    self.permission = ko.observable(contributor.permission);
-    self.curPermission = ko.observable(self.getPermission(self.permission()));
-    self.deleteStaged = ko.observable(contributor.deleteStaged || false);
-    self.removeContributor = 'Remove contributor';
+
+    self.deleteStaged = ko.observable(false);
+
     self.pageOwner = pageOwner;
+
     self.serialize = function() {
         return JSON.parse(ko.toJSON(self));
     };
 
     self.canEdit = ko.computed(function() {
-      return self.currentUserCanEdit && !self.isAdmin;
+        return self.currentUserCanEdit && !self.isAdmin;
     });
 
     self.remove = function() {
+        if (parent.visibleCount() > 0 && self.visible()) {
+            parent.visibleCount(parent.visibleCount() - 1);
+        }
+        if (parent.adminCount() > 0 && self.permission() === 'admin') {
+            parent.adminCount(parent.adminCount() - 1);
+        }
         self.deleteStaged(true);
     };
-    self.unremove = function(data, event) {
-        var $target = $(event.target);
-        if (!$target.hasClass('contrib-button')) {
+    self.unremove = function() {
+        if (self.deleteStaged()) {
             self.deleteStaged(false);
+            parent.visibleCount(parent.visibleCount() + self.visible());
+            parent.adminCount(parent.adminCount() + (self.permission() === 'admin'));
```

Comment:
Hmmm...this is adding an integer and a boolean and oddly enough `true == 1` and `false == 0` so it works. To make this clearer, can you change this check so that it does not rely on automatic type conversion?